plasmapy/particles/particle_class.py

```diff
@@ -2053,30 +2055,46 @@ class CustomParticle(AbstractPhysicalParticle):
     >>> custom_particle = CustomParticle(
     ...     mass=1.2e-26 * u.kg,
     ...     charge=9.2e-19 * u.C,
-    ...     symbol="Ξ",
     ... )
     >>> custom_particle.mass
     <Quantity 1.2e-26 kg>
     >>> custom_particle.charge
     <Quantity 9.2e-19 C>
-    >>> custom_particle.symbol
-    'Ξ'
-    >>> import pytest
-    >>> with pytest.warns(UserWarning):
-    ...    custom_particle = CustomParticle(
-    ...        mass=1.5e-26 * u.kg,
-    ...        charge=-1,
-    ...        symbol="Ξ",
-    ...    )
-    >>> custom_particle.mass
+    >>> average_particle = CustomParticle(
+    ...     mass=1.5e-26 * u.kg,
+    ...     Z = -1.5,
+    ...     symbol="Ξ",
+    ... )
+    >>> average_particle.mass
     <Quantity 1.5e-26 kg>
-    >>> custom_particle.charge
-    <Quantity -1.60217...e-19 C>
-    >>> custom_particle.symbol
+    >>> average_particle.charge
+    <Quantity -2.40326...e-19 C>
+    >>> average_particle.symbol
     'Ξ'
     """
 
-    def __init__(self, mass: u.kg = None, charge: (u.C, Real) = None, symbol=None):
+    def __init__(
+        self,
+        mass: u.kg = None,
+        charge: u.C = None,
+        symbol: Optional[str] = None,
+        *,
+        Z: Optional[Real] = None,
+    ):
+
+        if isinstance(charge, Real):
```

Comment:
```suggestion
        # TODO py3.10 replace ifology with structural pattern matching
        if isinstance(charge, Real):
```

It's not that far out, let's be prepared!

Reply:
Good point!  I wonder if this is something that these new automagical refactoring tools would be able to handle by that time.